generators/server/templates/gradle/_profile_dev.gradle

```diff
@@ -32,19 +32,16 @@ task gulpConstantDev(type: GulpTask) {
     args = ["ngconstant:dev", "--no-notification"]
 }<% } %>
 
-task setProdProperties(dependsOn: bootRun) << {
-    doFirst {
-        System.setProperty('spring.profiles.active', profiles)
-    }
-}
-
 processResources {
+    filesMatching('**/.build-properties.xml') {
+        filter {
+            it.replace('@spring.profiles.default@', profiles)
+        }
+    }
     filesMatching('**/logback-spring.xml') {
         filter {
             it.replace('@logback.loglevel@', logbackLoglevel)
         }
     }
 }
 
-<% if (!skipClient) { %>
-setProdProperties.dependsOn gulpConstantDev<% } %>
```

Comment:
this might affect cases when we `bootRun` dev profile after running a prod build, can you check this @erikkemperman gulpConstantDev task should be run may be we need to change this to `bootRun.dependsOn gulpConstantDev` cc @atomfrede